### PR TITLE
update lr_finder to check for attribute if not running fast_dev_run

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

more info here: https://github.com/PyTorchLightning/pytorch-lightning/pull/5638#discussion_r567815227
In short lr_find should mimic the behavior of scale_batch_size when searching for lr_attribute.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/tuner/lr_finder.py [hunk 1]
```diff
@@ -104,19 +90,21 @@ def lr_find(
 
         num_training: number of learning rates to test
 
-        mode: search strategy, either 'linear' or 'exponential'. If set to
-            'linear' the learning rate will be searched by linearly increasing
-            after each batch. If set to 'exponential', will increase learning
-            rate exponentially.
+        mode: Search strategy to update learning rate after each batch:
+
+            - ``'exponential'`` [default]: Will increase learning rate exponentially.
```
Comment (carmocca) [inline pytorch_lightning/tuner/lr_finder.py:95]: If i'm not mistaken, parentheses are generally used for this

```suggestion
            - ``'exponential'`` (default): Will increase the learning rate exponentially.
```

File: pytorch_lightning/tuner/lr_finder.py [hunk 2]
```diff
@@ -104,19 +90,21 @@ def lr_find(
...
+            - ``'linear'``: Will increate learning rate linearly.
```
Comment (carmocca) [inline pytorch_lightning/tuner/lr_finder.py:96]:
```suggestion
            - ``'linear'``: Will increase the learning rate linearly.
```

File: pytorch_lightning/tuner/lr_finder.py [hunk 3]
```diff
@@ -104,19 +90,21 @@ def lr_find(
...
 
         early_stop_threshold: threshold for stopping the search. If the
             loss at any point is larger than early_stop_threshold*best_loss
             then the search is stopped. To disable, set to None.
 
-        datamodule: An optional `LightningDataModule` which holds the training
+        datamodule: An optional ``LightningDataModule`` which holds the training
             and validation dataloader(s). Note that the `train_dataloader` and
```
Comment (carmocca) [inline pytorch_lightning/tuner/lr_finder.py:103]:
```suggestion
            and validation dataloader(s). Note that the ``train_dataloader`` and
```

File: pytorch_lightning/tuner/lr_finder.py [hunk 4]
```diff
@@ -200,6 +192,14 @@ def lr_find(
     if trainer.progress_bar_callback:
         trainer.progress_bar_callback.enable()
 
+    # Update lr attr if required
+    if update_attr:
+        lr = lr_finder.suggestion()
+
+        # TODO: log lr.results to self.logger
+        lightning_setattr(model, lr_attr_name, lr)
+        log.info(f'Learning rate set to {lr}')
```
Comment (carmocca) [inline pytorch_lightning/tuner/lr_finder.py:201]: rank zero info?

**Review by carmocca — APPROVED** [review]
Minor comments. Nice job!

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/tuner/lr_finder.py (hunk 4 above)
Comment (rohitgr7) [reply pytorch_lightning/tuner/lr_finder.py:201]: I guess it is not running inside any process, so not required?

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**